docs/psoc6/quickref.rst

```diff
@@ -669,10 +669,12 @@ I2S bus
 
 See :ref:`machine.I2S <machine.I2S>`. 
 
-Supported sample rates are 8KHz, 16KHz, 32KHz, 48KHz, 22.05KHz, 44.1KHz.
+.. note::
+    The only difference with the reference API is the usage of the ``id`` in the constructor ``IS2()``. In the PSoC6 port the ``id`` value is ignored. 
```

Comment:
Here this sentence gives a overview that no value of id is required no? 
According to the main I2S doc 
"id identifies a particular I2S bus; it is board and port specific". 
 Then in our case instead of bus that is used for I2S instance allocation. So the id is not ignored right? It's having a use. So whether that statement is required?

Reply:
The argument is required because the extmod/machine_i2s.c is providing the constructor API, and the "id" is mandatory.
In our port we have not implemented an [id -> (clk, ws, sd)] map. That would allow the user to avoid passing the pins, simplifying the constructor call (at our cost of implementing and maintain these maps for each machine class).

This i2s object "id" param is as well expected in the struct by the extmod/machine_i2s.c.  So we are using it to avoid searching the current instance in the array of i2s instances during deinit(). But we could also just left it unused. Anyhow, this is private, irrelevant at MicroPython API level.
They just pass the pins, and they get an instance of I2S. The constructor will fail (from resource allocation point of view) due to: A) no more instances available B) cyhal complains of already allocated pins for i2s (or any other GPIO dependent class). 

Let´s discuss later today how to document this that is clearer.